### PR TITLE
Proof-of-Concept for testing code samples

### PR DESCRIPTION
**Pull request by zupo**
There are a number of code samples throughout nix.dev. How do we know they still work? We don't!

This commit introduces a way for us to extract these code samples into files and then run tests against them in CI. This will hopefully help us catch regressions in future updates to nix, NixOS and/or this guide.

Additionally, I included a darwin specific nix-shell configuration that I use personally on my M1 Mac to work on this repo. Might be useful for someone.

### CODE REVIEW

**Comment by zupo** [comment]
@domenkozar: can you update the flakes-based nix-shell to have python 3.9? I have Python3.9 in my shell-darwin.nix and Ideally I'd keep it because I need the 3.9 feature of combining two dicts with `|` operator. 

If the update of flakes to have python3.9 is annoying, I can rewrite that code to be python3.8 compatible.

**Comment by domenkozar** [comment]
https://github.com/nix-dot-dev/nix.dev/pull/249

**Comment by domenkozar** [comment]
from cloudflare logs:

```


15:57:47.773 |  
-- | --
15:57:47.773 | Exception occurred:
15:57:47.773 | File "/opt/buildhome/repo/source/_ext/extractable_code_block.py", line 30, in ExtractableCodeBlock
15:57:47.773 | "extract_dir": directives.path,
15:57:47.773 | TypeError: unsupported operand type(s) for \|: 'dict' and 'dict'
15:57:47.774 | The full traceback has been saved in /tmp/sphinx-err-s5msp6gk.log, if you want to report the issue to the developers.
15:57:47.774 | Please also report this if it was a user error, so that a better error message can be provided next time.
15:57:47.774 | A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
15:57:47.844 | Makefile:53: recipe for target 'html' failed
15:57:47.844 | make: *** [html] Error 2
15:57:47.852 | Failed: build command exited with code: 2


```

File: shell-darwin.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# @zupo uses this file to work on nix.dev on his M1 Monterey
```
Comment (domenkozar) [inline shell-darwin.nix:1]: Note that this isn't tested on the CI, so it will likely break with updates

**Review by domenkozar — APPROVED** [review]

File: shell-darwin.nix (hunk 1 above)
Comment (zupo) [reply shell-darwin.nix:1]: Yeah, that's fine, it's mostly for me.

**Pull request merged**